drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -376,7 +376,9 @@ def user(id:)
     def merge_audit_history(filters: nil)
       raise 'Access denied' unless current_user.can_merge_clients?
 
-      scope = Hmis::ClientMergeAudit.all
+      # can_merge_clients? is typically a global permission, but just in case, filter to clients viewable by the current user.
+      # This prevents the UI from crashing if the MergeAuditEvent GraphQL object is trying to resolve a client the user doesn't have permission to see.
+      scope = Hmis::ClientMergeAudit.viewable_by(current_user)
```

Comment:
suggest clarifying/making more succinct:
```suggestion
      # can_merge_clients is a global permission (not limited to specific clients), but MergeAuditEvent resolves
      # the client record which requires object-level authorization. Filter by viewable_by to prevent authorization failures.
      scope = Hmis::ClientMergeAudit.viewable_by(current_user)
```